src/industry_gui.cpp

```diff
@@ -1544,7 +1555,7 @@ class IndustryDirectoryWindow : public Window {
 	}
 
 public:
-	IndustryDirectoryWindow(WindowDesc *desc, WindowNumber number) : Window(desc)
+	IndustryDirectoryWindow(WindowDesc *desc, WindowNumber number) : Window(desc), industry_editbox(MAX_FILTER_LENGTH)
```

Comment:
Take into account the character length, just like in your other pr. :)

```suggestion
	IndustryDirectoryWindow(WindowDesc *desc, WindowNumber number) : Window(desc), industry_editbox(MAX_FILTER_LENGTH * MAX_CHAR_LENGTH, MAX_FILTER_LENGTH)
```

Reply:
That's nice! changing the description of the variable to char rather than bytes.
Note that there is no constant that determines the limit for industry name as they exist for other stuff like town names, station names, etc.